src/zepben/evolve/streaming/grpc/connect.py

```diff
@@ -22,17 +22,25 @@
 
 class AuthTokenPlugin(grpc.AuthMetadataPlugin):
 
-    def __init__(self, host, conf_address, client_id, client_secret):
+    def __init__(self, host, conf_address=None, client_id=None, client_secret=None, authenticator=None):
         self.host = host
         self.conf_address = conf_address
         self.client_id = client_id
         self.client_secret = client_secret
-        self.token = ""
         self.token_expiry = 0
-        self._refresh_token()
+        self.authenticator = authenticator
+        if self.authenticator:
+            self.token = f"Bearer {self.authenticator.get_token()}"
+        elif conf_address and client_id and client_secret:
+            self.token = ""
+            self._refresh_token()
+        else:
+            raise AuthException("Please provide authenticator or all of conf_address, client_id and client_secret.")
 
     def __call__(self, context, callback):
-        if datetime.utcnow().timestamp() > self.token_expiry:
+        if self.authenticator:
+            self.token = f"Bearer {self.authenticator.get_token()}"
+        if self.conf_address and datetime.utcnow().timestamp() > self.token_expiry:
```

Comment:
this should probably be an elif, just in case someone provides conf_address and an authenticator.